app/__init__.py

```diff
@@ -1,110 +1,121 @@
-    password = request.json.get("password")
-    current_user = []
-
-    for user in users:
-        if user["username"] == username:
-            current_user.append(user)
-    if len(current_user) != 0:
-        if current_user[0]["password"] == password:
+    if not request.is_json:
+        return jsonify({"msg": "Missing JSON in request"}), 400
+    _username = request.json.get("username", None)
+    _password = request.json.get("password", None)
+
+    user = model.get_user(_username)
+
+    if user is not None:
+        if user["username"] == _username and user["password"] == _password:
             """Create user session"""
-            session["user_id"] = current_user[0]["user_id"]
-            return jsonify({"message": "Sign in Successful!"}), 202
-        else:
-            return jsonify({"message": "Wrong username or password"}), 401
+            access_token = create_access_token(identity=_username)
+            return jsonify({"message": "Sign in Successful!", "token": access_token}), 202
+        return jsonify({"message": "Wrong username or password"}), 401
     return jsonify({"message": "User Not Found"}), 404
 
 
-# Sign out User
-@app.route('/api/v1/users/signout', methods=['POST'])
-def signout_user():
-    """Check if session is empty"""
-    if len(session) == 0:
-        return jsonify({"message": "You must be Signed in to Sign out"}), 403
-    session.pop("user_id", None)
-    return jsonify({"message": "Sign out Successful"}), 200
+# # Sign out User
+# @app.route('/api/v1/users/signout', methods=['POST'])
+# @jwt_required
+# def signout_user():
+#     user = current_user
+#     user.authenticated = False
+
+#     """Check if session is empty"""
+#     print(session)
+#     if len(session) == 0:
+#         return jsonify({"message": "You must be Signed in to Sign out"}), 403
+#     session.pop("username", None)
+#     return jsonify({"message": "Sign out Successful"}), 200
 
 
 # POST a request
 @app.route('/api/v1/users/requests', methods=['POST'])
+@jwt_required
 def create_request():
-    if len(session) != 0:
-        user_id = session["user_id"]
-        request_data = {
-            "id": str(uuid.uuid4()),
-            "user_id": user_id,
-            "title": request.json.get("title"),
-            "type": request.json.get('type'),
-            "description": request.json.get("description"),
-            "category": request.json.get('category'),
-            "area": request.json.get('area')
-        }
-        requests.append(request_data)
-        return jsonify({"message": request_data}), 201
-    return jsonify({"message": "Sign In to make a requests"}), 401
+    current_user = get_jwt_identity()
+    print(current_user)
+    request_data = UserRequest(
+        request.json.get("title"),
+        request.json.get('_type'),
+        request.json.get("description"),
+        request.json.get('category'),
+        request.json.get('area')
+    )
+    request_data.create_request()
```

Comment:
make sure you validate this data before you save it in your db

Reply:
I've validated the data for create request and update request, including signup user